### ISSUE TITLE
pidstat-convert needs to ensure pid file names are not longer than 255 characters

### ISSUE THREAD

**Issue by portante** [distributed-system-analysis/pbench#1164]
`pidstat-convert` needs to ensure pid file names are not longer than 255 characters.

Example failure case:

```
# cat pidstat-stderr.txt
Could not open file /var/lib/pbench-agent/baseline_test_RHEL76_RHCS4_upstream_nautilus_2019-04-10T12-57-25-UTC/results/00000000/id-7670446295847779895/tools-default/pidstat/pids/633861/SCREEN -dmS pbench-tool-iostat-default-1 ->opt->pbench-agent->tool-scripts->iostat --start --iteration=1 --group=default --dir=->var->lib->pbench-agent->baseline_test_RHEL76_RHCS4_upstream_nautilus_2019-04-10T12-57-25-UTC->results->00000000->id-7670446295847779895 default --interval=30 at /opt/pbench-agent/tool-scripts/postprocess/pidstat-convert line 85, <> line 656.
```

If you count the above characters you will arrive at 452.

Suggest we calculate the MD5 sum for command line of a give pid, and use that as the filename on disk, adding a `<md5sum>.cmd` file containing the original command line.

**Comment by atheurer** [comment]
I think we end up truncating the process&args later because we support a data series name of limited length anyway, so it might be possible to just use that truncated process&arg in the filename.  I assume the path is not part of the problem, and it's just the, "SCREEN -dmS pbench-tool-iostat-default-1 ->opt->pbench-agent->tool-scripts->iostat --start --iteration=1 --group=default --dir=->var->lib->pbench-agent->baseline_test_RHEL76_RHCS4_upstream_nautilus_2019-04-10T12-57-25-UTC->results->00000000->id-7670446295847779895 default --interval=30"

**Comment by portante** [comment]
Won't using the truncated name raise the number of possible conflicts with file names if pids get re-used?

Seems like MD5 is really simple solution and always maps "effectively" 1-to-1.

**Comment by atheurer** [comment]
Good point, although I think the risk is having more than one of the exact same commands.  I agree MD5 should resolve all of this.

**Comment by ndokos** [comment]
Issue #1169 is a duplicate of this.

**Comment by portante** [comment]
@atheurer, thanks for taking a look at how to fix this.

**Comment by chaitanyaenr** [comment]
FYI we are planning to use v0.58 pbench bits for the OCP 4.1 large scale testing next week, we will bake in the fix into the images once it lands. Thanks.
